### ISSUE TITLE
Support custom syntax color type

### ISSUE THREAD

**Issue by simonbs** [louisdh/savannakit#5]
The token types support for highlightning is currently limited to the cases exposed by the SyntaxColorType but for some programming languages it would be beneficial if SavannaKit support more/other types.

**Comment by simonbs** [comment]
I took a stab at this but I'm not totally sure that my approach is the best. I've changed `SyntaxColorType` to be a struct with an identifier rather than an enum. I'm not sure an enum is the right approach for a framework that aims to be language agnostic.

The cases specified in the current implementation of `SyntaxColorType` may be suitable for most programming languages but if you're writing a markdown editor, most of the cases probably don't make sense. And while the cases _probably_ are suitable, it's not unlikely that more/other types are needed for different programming language.
Therefore I think it may be alright to use a simple string identifier instead, allowing the user of SavannaKit to specify arbitrary syntax color types.

I see two downsides to this approach:
1. When returning the color in `color(for:)`, there's a need to handle the awkward "default" cause where the theme is asked to provide a color for an unsupported identifier. In practice, this won't happen.
2. Editor placeholder is currently tied to the `editorPlaceholder` case of the enum. To accommodate this, I've added an `isEditorPlaceholder` property to the new struct. Any syntax color type can now be an editor placeholder.

My suggested implementation can be seen in https://github.com/louisdh/savannakit/pull/15. I'm very open to suggestions on how this can be improved.
